File: backend/src/main/java/com/back/domain/applicant/dto/response/ApplicantResponseDto.java
package com.back.domain.applicant.dto.response;

import com.back.domain.applicant.entity.Applicant;
import lombok.Builder;

@Builder
public record ApplicantResponseDto (
    Long id,
    String name,
    String phone,
    String email,
    String address
) {
    public static ApplicantResponseDto from(Applicant applicant) {
        return ApplicantResponseDto.builder()
                .id(applicant.getId())
                .name(applicant.getName())
                .phone(applicant.getPhone())
                .email(applicant.getEmail())
                .address(applicant.getAddress())
                .build();
    }
}

File: backend/src/main/java/com/back/domain/applicant/dto/response/ApplicantResponseDto.kt
package com.back.domain.applicant.dto.response

import com.back.domain.applicant.entity.Applicant

data class ApplicantResponseDto(
    val id: Long?,
    val name: String,
    val phone: String,
    val email: String,
    val address: String
) {
    companion object {
        fun from(applicant: Applicant): ApplicantResponseDto? {
            return ApplicantResponseDto(
                id = applicant.id,
                name = applicant.name,
                phone = applicant.phone,
                email = applicant.email,
                address = applicant.address
            )
        }
    }
}
